### PR TITLE
🐛 Fix display when building a monolith

### PR DESCRIPTION
**Pull request by liam-teale**
#### Summary:
Fix display when a project is built as a monolith

#### Motivation:
When you build a PROS project as a monolith, the display is unusable. This is because the weakly declared functions in `libpros.a` do not get overridden with the strongly declared functions in `liblvgl.a`, but only if building a monolith.

#### How does this fix work?
Weakly defined symbols are now only declared if LVGL is not present.
Except for one:
the `pros_initialize` function calls the weakly declared `display_initialize()` function. One solution would be to call this function after `pros_initialize`, in liblvgl.a. But alas, this won't work (see https://github.com/purduesigbots/pros/pull/144/#issuecomment-496901942). Instead, we initialize pros partially, then `display_initialize` is called as a constructor, then pros is initialized fully.

#### The underlying issue

The underlying issue is that weakly defined symbols do not get overridden, but only if building a monolith. Despite my best efforts, I was not able to identify the cause. It was not the order in which the libraries were linked, or the objects getting optimized out, since in my tests I eliminated both cases and the weakly declared functions still weren't overridden.

This PR does not fix this problem, it just treats the symptoms with a workaround.

#### Why does it work if using hot/cold linking?

Because all objects in every library are included in the cold package. When using a monolith, only the required objects are included. We could do the same when building a monolith, but this would cause the binary size to grow in size by an order of magnitude. Hence, this PR introduces a different workaround.

#### Incompatibilities

This PR requires a new version of liblvgl to be created. Current versions of liblvgl are not compatible with this version of pros. However, the new version of liblvgl will not be compatible with current versions of PROS. The new version of LVGL and the new version of PROS must be released simultaneously, and PROS would require, at minimum, a minor version increment (NOT a patch).

liblvgl pull request: https://github.com/purduesigbots/liblvgl/pull/51

See the related PR to liblvgl: 

#### Test Plan (pending on PROS team sharing a template zip):

This snippet works as expected when user code is built as
-[ ] hot/cold packages
-[ ] monolith

```c++
#include "main.h"

void initialize() {
    pros::lcd::initialize();
}
```

### CODE REVIEW

File: include/pros/llemu.h [hunk 1]
```diff
@@ -15,7 +15,7 @@
 /******************************************************************************/
 #ifdef _PROS_INCLUDE_LIBLVGL_LLEMU_H
 #include "liblvgl/llemu.h"
-#endif
+#else
```
Comment (djava) [inline include/pros/llemu.h]: add a comment explaining this fix

File: include/pros/llemu.h [hunk 2]
```diff
@@ -51,4 +51,6 @@ bool __attribute__((weak)) lcd_print(int16_t line, const char* fmt, ...)  {
 } // extern "C"
 #endif//__cplusplus
 
+#endif // _PROS_INCLUDE_LIBLVGL_LLEMU_H
```
Comment (djava) [inline include/pros/llemu.h]:
```suggestion
#endif // _PROS_INCLUDE_LIBLVGL_LLEMU_H not present
```

File: src/system/startup.c [hunk 3]
```diff
@@ -33,21 +32,24 @@ extern void invoke_install_hot_table();
 // gives the compiler instructions on the priority of the constructor,
 // from 0-~65k. The first 0-100 priorities are reserved for language
 // implementation.
-__attribute__((constructor(101))) static void pros_init(void) {
+__attribute__((constructor(101))) static void pros_init_a(void) {
```
Comment (djava) [inline src/system/startup.c]: maybe `_1` and `_2` suffix instead of `_a` and `_b`? I was initially wondering what `_a` stood for.

**Comment by Rocky14683** [comment]
https://artprodcus3.artifacts.visualstudio.com/A268c8aad-3bb0-47d2-9a57-cf06a843d2e8/a421d041-2579-4bf8-a0aa-d7a0307dade3/_apis/artifact/cGlwZWxpbmVhcnRpZmFjdDovL3B1cmR1ZS1hY20tc2lnYm90cy9wcm9qZWN0SWQvYTQyMWQwNDEtMjU3OS00YmY4LWEwYWEtZDdhMDMwN2RhZGUzL2J1aWxkSWQvNjYwMy9hcnRpZmFjdE5hbWUvdGVtcGxhdGU1/content?format=zip

File: src/system/startup.c (hunk 3 above)
Comment (WillXuCodes) [reply src/system/startup.c]: a more verbose `stage_1` and `stage_2` would do the trick too

**Comment by liam-teale** [comment]
Suggestions implemented, merge conflicts fixed. Ready for another round of reviews

**Comment by Rocky14683** [comment]
It's still giving me data abort

**Comment by liam-teale** [comment]
are you using the right version of LVGL?

**Comment by liam-teale** [comment]
Closing, I'm not happy with this PR. It should solve the underlying issue, not work around it.

**Pull request closed without merging**